src/CachePlugin.php

```diff
@@ -225,6 +224,10 @@ private function getCacheControlDirective(ResponseInterface $response, $name)
      */
     private function createCacheKey(RequestInterface $request)
     {
+        if ('POST' === $request->getMethod()) {
+            return hash($this->config['hash_algo'], $request->getMethod().' '.$request->getUri().' '.$request->getBody());
```

Comment:
i don't like that we specifically do this for POST (as i want to support configuring all request methods as cacheable). i think using the body in hash should not be an issue - its usually just empty. technically, GET requests can have a body as well, for example with elasticsearch. so its actually an improvement to also use the body for the hash.